src/Raven.Server/Https/ExternalCertificateValidator.cs

```diff
@@ -0,0 +1,288 @@
+                {
+                    task.Start();
+
+                    if (_externalCertificateValidationCallbackCache.Count > 50)
+                    {
+                        foreach (var item in _externalCertificateValidationCallbackCache.Where(x => x.Value.IsCompleted).OrderBy(x => x.Value.Result.Until).Take(25))
+                        {
+                            _externalCertificateValidationCallbackCache.TryRemove(item.Key, out _);
+                        }
+                    }
+                }
+            }
+
+            CachedValue cachedValue;
+            try
+            {
+                cachedValue = existingTask.Result;
+            }
+            catch
+            {
+                _externalCertificateValidationCallbackCache.TryRemove(cacheKey, out _);
+                throw;
+            }
+
+            if (_server.Time.GetUtcNow() < cachedValue.Until)
+                return cachedValue.Valid;
+
+            var cachedValueNext = cachedValue.Next;
+            if (cachedValueNext != null)
+                return ReturnTaskValue(cachedValueNext);
+
+            task = new Task<CachedValue>(() =>
+                CheckExternalCertificateValidation(senderHostname, certificate, chain, sslPolicyErrors, log));
+
+            var nextTask = Interlocked.CompareExchange(ref cachedValue.Next, task, null);
+            if (nextTask != null)
+                return ReturnTaskValue(nextTask);
+
+
+            task.ContinueWith(done =>
+            {
+                _externalCertificateValidationCallbackCache.TryUpdate(cacheKey, done, existingTask);
+            }, TaskContinuationOptions.OnlyOnRanToCompletion);
+
+            task.Start();
+
+            return cachedValue.Valid; // we are computing this, but may take some time, let's use cached value for now
+
+            bool ReturnTaskValue(Task<CachedValue> task)
+            {
+                if (task.IsCompletedSuccessfully)
+                    return task.Result.Valid;
+
+                // not done yet? return the cached value
+                return cachedValue.Valid;
+            }
+        }
+
+        // https://stackoverflow.com/questions/14241479/how-to-split-a-space-delimited-list-of-paths-where-paths-can-include-spaces-in
+        private static readonly Regex re = new Regex(@"^([ ]*((?<r>[^ ""]+)|[""](?<r>[^""]*)[""]))*[ ]*$");
+
+        private static IEnumerable<string> RegexSplit(string input)
+        {
+            var m = re.Match(input ?? "");
+            if (!m.Success)
+                throw new ArgumentException("Malformed input.");
+
+            return from Capture capture in m.Groups["r"].Captures select capture.Value;
+        }
+
+        private long GetScriptLastWriteHash()
+        {
+            long hash = File.GetLastWriteTimeUtc(_server.Configuration.Security.CertificateValidationExec).Ticks;
+            var args = RegexSplit(_server.Configuration.Security.CertificateValidationExecArguments);
```

Comment:
Do we need args here? You cannot set them more than once, during server startup?

Reply:
Fixing